### PR TITLE
Full sleep wait flex counter polling thread when POLL_COUNTER_STATUS is disable

### PR DESCRIPTION
**Pull request by wendani**
A PR tailored for 201811 based on https://github.com/Azure/sonic-sairedis/pull/462   

In 201811, we do not have buffer pool watermark and rif counters. So this version resolves conflicts raised herein when backporting to 201811.

### CODE REVIEW

File: syncd/syncd_flex_counter.cpp [hunk 1]
```diff
@@ -548,13 +587,16 @@ void FlexCounter::removeCounterPlugin(
...
+    std::unique_lock<std::mutex> lkMgr(fc.m_mtx);
+
     fc.m_queuePlugins.clear();
     fc.m_portPlugins.clear();
     fc.m_priorityGroupPlugins.clear();
 
     // Remove flex counter if all counter IDs and plugins are unregistered
     if (fc.isEmpty())
     {
+        lkMgr.unlock();
```
Comment (kcudnik) [inline syncd/syncd_flex_counter.cpp]: please don't do manual unlocking, this is really bad design, use scope lock/unlock

File: syncd/syncd.cpp [hunk 2]
```diff
@@ -2782,9 +2782,11 @@ void processFlexCounterGroupEvent(
 {
     SWSS_LOG_ENTER();
 
-    std::lock_guard<std::mutex> lock(g_mutex);
     swss::KeyOpFieldsValuesTuple kco;
-    consumer.pop(kco);
+    {
```
Comment (kcudnik) [inline syncd/syncd.cpp]: why you moved this here? please revert this change, since g_mutex is not protecting only consumer.pop but other variables too!!

**Comment by wendani** [comment]
Discussed offline with Renuka. The problem is in the signal handling path. Not the normal path that we have thought of.

**Comment by kcudnik** [comment]
there is no signal handling in syncd, also there is no way to handle sigkill

**Comment by renukamanavalan** [comment]
SIGKILL can't be handled in any case.
 The easiest fix could be, add an exit handler and all that the exit handler does is call abort()
